src/main/java/com/intellij/rt/debugger/agent/CaptureStorage.java

```diff
@@ -383,9 +396,53 @@ public int getRecursionDepth() {
   }
 
   // to be run from the debugger
+
+  /**
+   * Returns the captured stack trace of the current thread.
+   */
   @SuppressWarnings("unused")
   public static String getCurrentCapturedStack(int limit) throws IOException {
-    return wrapInString(CURRENT_STACKS.get().peekLast(), limit);
+    return wrapInString(getStacksForCurrentThread().peekLast(), limit);
+  }
+
+  /**
+   * If storing stack traces for all threads is enabled (`debugger.async.stack.trace.for.all.threads` is true),
+   * returns the captured stack trace of the given thread or null if no stack trace was not captured.
+   *
+   * If `debugger.async.stack.trace.for.all.threads` is false,
+   * it only returns the captured stack trace for the current thread and null if no stack trace was captured or if the given thread is not the current thread.
+   */
+  @SuppressWarnings("unused")
+  public static String getCapturedStackForThread(int limit, Thread thread) throws IOException {
+    Deque<CapturedStack> capturedStacks = storeAsyncStackTracesForAllThreads
+            ? THREAD_TO_STACKS_MAP.get(thread)
+            : (thread == Thread.currentThread() ? CURRENT_STACKS.get() : null);
+    if (capturedStacks == null) return null;
+    return wrapInString(capturedStacks.peekLast(), limit);
+  }
+
+  /**
+   * If storing stack traces for all threads is enabled (`debugger.async.stack.trace.for.all.threads` is true),
+   * returns stack traces of all threads as a map of a thread to it's captured stack trace or an empty map if no stack traces were captured.
+   *
+   * If `debugger.async.stack.trace.for.all.threads` is false,
+   * only returns the captured stack trace of the current thread or an empty map if no stack trace was captured for the current thread.
+   */
+  public static Map<Thread, String> getAllCapturedStacks(int limit) throws IOException {
+    HashMap<Thread, String> threadToStack = new HashMap<>();
+    if (storeAsyncStackTracesForAllThreads) {
+      for (Thread thread : THREAD_TO_STACKS_MAP.keySet()) {
```

Comment:
And here we should also filter dead threads.

Reply:
Fixed this